src/convertToObject.js

```diff
@@ -6,7 +6,48 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrSoursString = sourceString;
+  const result = {};
+
+  for (let i = 0; i < arrSoursString.length; i++) {
+    if (arrSoursString[i] === ':') {
+      result[`${suchKey(arrSoursString, i)}`] = such(arrSoursString, i);
+    }
+  }
```

Comment:
Using a `for` loop to iterate over the string is not recommended for this task. Consider using string methods like `split` and iteration methods like `map` and `reduce` to transform the string into an object.